apps/omg_eth/lib/eth/root_chain.ex

```diff
@@ -236,24 +234,38 @@ defmodule OMG.Eth.RootChain do
   end
 
   def deposit_blknum_from_receipt(receipt) do
-    [{_, deposit_blknum, _, _}] = filter_receipt_events(receipt["logs"], "Deposit(address,uint256,address,uint256)")
+    [%{blknum: deposit_blknum}] =
+      filter_receipt_events(receipt["logs"], "Deposit(address,uint256,address,uint256)", &decode_deposit/1)
+
     deposit_blknum
   end
 
-  @spec filter_receipt_events([%{topics: [binary], data: binary()}], binary) :: [tuple]
-  def filter_receipt_events(receipt_logs, signature) do
+  defp decode_deposit(log) do
+    non_indexed_keys = [:currency, :amount]
+    non_indexed_key_types = [:address, {:uint, 256}]
+    indexed_keys = [:owner, :blknum]
+    indexed_keys_types = [:address, {:uint, 256}]
+
+    parse_events_with_indexed_fields(log, {non_indexed_keys, non_indexed_key_types}, {indexed_keys, indexed_keys_types})
+  end
+
+  defp decode_exit(log) do
+    non_indexed_keys = [:currency, :amount]
+    non_indexed_key_types = [:address, {:uint, 256}]
+    indexed_keys = [:owner, :utxo_pos]
+    indexed_keys_types = [:address, {:uint, 256}]
+
+    parse_events_with_indexed_fields(log, {non_indexed_keys, non_indexed_key_types}, {indexed_keys, indexed_keys_types})
+  end
+
+  @spec filter_receipt_events([%{topics: [binary], data: binary()}], binary, (map() -> map())) :: [map()]
+  def filter_receipt_events(receipt_logs, signature, decode_log) do
```

Comment:
I think I'd have a slight preference towards running `Enum.map(&decode_deposit/1)` in the function that calls this, instead of passing a function parameter, to keep simple.

Also I think this can be `defp` now, perhaps?